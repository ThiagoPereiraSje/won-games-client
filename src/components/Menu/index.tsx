import { Menu2 as MenuIcon } from 'styled-icons/remix-fill/index'
import {
  ShoppingCart as ShoppingCartIcon,
  Search as SearchIcon
} from 'styled-icons/material-outlined/index'

import Logo from 'components/Logo'

import * as S from './styles'

const Menu = () => {
  return (
    <S.Wrapper>
      <S.IconWrapper>
        <MenuIcon />
      </S.IconWrapper>

      <S.LogoWrapper>
        <Logo hideOnMobile />
      </S.LogoWrapper>

      <S.MenuGroup>
        <S.IconWrapper>
          <SearchIcon />
        </S.IconWrapper>

        <S.IconWrapper>
          <ShoppingCartIcon />
        </S.IconWrapper>
      </S.MenuGroup>
    </S.Wrapper>
  )
}

export default Menu
