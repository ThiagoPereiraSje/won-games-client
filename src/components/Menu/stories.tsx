import { ComponentMeta, ComponentStory } from '@storybook/react'
import Menu from '.'

export default {
  title: 'Menu',
  component: Menu
} as ComponentMeta<typeof Menu>

export const Basic: ComponentStory<typeof Menu> = (args: any) => (
  <Menu {...args} />
)
