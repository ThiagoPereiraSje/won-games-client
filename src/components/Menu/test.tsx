import { render, screen } from '@testing-library/react'
import Menu from '.'

describe('<Menu />', () => {
  it('should render the heading', () => {
    render(<Menu />)

    const element = screen.getByRole('heading', { name: /Menu/i })

    expect(element).toBeInTheDocument()
  })

  it('should render correctly', () => {
    const { container } = render(<Menu />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
