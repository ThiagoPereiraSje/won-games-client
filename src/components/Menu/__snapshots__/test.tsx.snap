// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render correctly 1`] = `
<main
  class=""
>
  <h1>
    Hello Menu
  </h1>
</main>
`;
